akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala

```diff
@@ -304,6 +305,66 @@ class Http2ClientSpec extends AkkaSpecWithMaterializer("""
       }
 
     }
+
+    "support for configurable ping" should {
+      "send pings when there is an active but slow stream from client" in new TestSetup with NetProbes with Http2FrameHpackSupport {
+        override def settings = {
+          val default = super.settings
+          default.withHttp2Settings(default.http2Settings.withPingInterval(2.seconds).withPingTimeout(1.second))
+        }
+        val streamId = 0x1
+        val requestStream = TestPublisher.probe[ByteString]()
+        emitRequest(streamId, HttpRequest(
+          protocol = HttpProtocols.`HTTP/2.0`,
+          entity = HttpEntity(ContentTypes.`application/octet-stream`, Source.fromPublisher(requestStream))))
+        expectDecodedHEADERS(streamId, endStream = false)
+        Thread.sleep(2000) // no data for this interval should trigger ping
```

Comment:
Maybe `expectNoBytes(2.seconds)` instead?

Reply:
Yes, simple 2.seconds wont work because the interval (and timeout) is from sending on the other side, not from receiving the frame. So in reality I rather had to do something like you describe Ignasi.